url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/imex/FisExporter.java

```diff
@@ -0,0 +1,527 @@
+        for (OutputVariable outputVariable : engine.getOutputVariables()) {
+            Defuzzifier other = outputVariable.getDefuzzifier();
+            if (defuzzifier == null) {
+                defuzzifier = other;
+            } else if (!defuzzifier.getClass().equals(other.getClass())) {
+                throw new RuntimeException("[export error] "
+                        + "fis files require all output variables "
+                        + "to have the same defuzzifier");
+            }
+        }
+        return toString(defuzzifier);
+    }
+
+    protected String translate(List<Proposition> propositions, List<Variable> variables) {
+        StringBuilder result = new StringBuilder();
+        for (Variable variable : variables) {
+            int termIndexPlusOne = 0;
+            double plusHedge = 0;
+            int negated = 1;
+            for (Proposition proposition : propositions) {
+                if (!variable.equals(proposition.getVariable())) {
+                    continue;
+                }
+                for (int termIndex = 0; termIndex < variable.numberOfTerms(); ++termIndex) {
+                    if (variable.getTerm(termIndex).equals(proposition.getTerm())) {
+                        termIndexPlusOne = termIndex + 1;
+                        break;
+                    }
+                }
+                if (proposition.getHedges().size() > 1) {
+                    FuzzyLite.logger().warning("[export warning] "
+                            + "only a few combinations of multiple "
+                            + "hedges are supported in fis files");
+                }
+                for (Hedge hedge : proposition.getHedges()) {
+                    if (hedge instanceof Not) {
+                        negated *= -1;
+                    } else if (hedge instanceof Seldom) {
+                        plusHedge += 0.01;
+                    } else if (hedge instanceof Somewhat) {
+                        plusHedge += 0.05;
+                    } else if (hedge instanceof Very) {
+                        plusHedge += 0.2;
+                    } else if (hedge instanceof Extremely) {
+                        plusHedge += 0.3;
+                    } else if (hedge instanceof Any) {
+                        plusHedge += 0.99;
+                    } else {
+                        plusHedge = Double.NaN; // Unreconized hedge combination
+                    }
+                }
+                break;
+            }
+
+            if (negated < 0) {
+                result.append("-");
+            }
+            if (!Double.isNaN(plusHedge)) {
+                result.append(Op.str(termIndexPlusOne + plusHedge));
+            } else { //Unreconized hedge combination
+                result.append(String.format("%d.?", termIndexPlusOne));
+            }
+            result.append(" ");
+        }
+        return result.toString();
+    }
+
+    public String toString(Term term) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 361-370 into a separate method.